packages/hierarchies/src/hierarchies/HierarchyNode.ts

```diff
@@ -26,6 +26,8 @@ export type HierarchyNodeFilteringProps = {
   hasFilterTargetAncestor?: boolean;
   /** Paths to node's children that are filter targets. */
   filteredChildrenIdentifierPaths?: HierarchyFilteringPath[];
+  /** Paths index in all filter paths */
+  filteredChildrenIdentifierPathsIndex?: number[];
```

Comment:
I don't understand the meaning of this. Can you explain in more detail?

Reply:
This tells which original nodeIdentifierPath is filteredChildrenIdentifierPaths pointing to.
For example if we create a `FilteringHierarchyDefinition` with these `nodeIdentifierPaths`: 
```ts
nodeIdentifierPaths: [
  [a, b, c],
  [a, c, d],
  [a, b, d]
]
```
And try to get `filteredChildrenIdentifierPaths` for node b, 
The following `filteredChildrenIdentifierPaths` would be created:
```ts
filteredChildrenIdentifierPaths: [
  [c],
  [d]
]
```
And the following `filteredChildrenIdentifierPathsIndex` would be created:
```ts
filteredChildrenIdentifierPathsIndex: [0, 2]
```
filteredChildrenIdentifierPathsIndex tells us that first path ([c]) in `filteredChildrenIdentifierPaths` is created from 0th path ([a, b, c]) of `nodeIdentifierPaths`. And the the second path ([d]) is created from 2nd path ([a, b, d]).

Reply:
After talking in private, changed the implementation, this is no longer needed.